libs/api/books/src/lib/reading-list.service.ts

```diff
@@ -28,4 +28,14 @@ export class ReadingListService {
       return list.filter(x => x.bookId !== id);
     });
   }
+
+  async markBookAsFinished(id: string, item: ReadingListItem): Promise<void> {
+    this.storage.update(list => {
+      return list.filter(listItem => listItem.bookId !== id);
+      const itemToBeMarked = list.find( listItem => listItem.bookId === id);
```

Comment:
You've got unreachable code here...this means that, while the front end may look ok because of the optimistic updating, the backend never gets the update.

Reply:
Made updated to code to make sure no unreachable code is present.